src/ios/CDVLocation.m

```diff
@@ -94,61 +94,62 @@ - (BOOL)isLocationServicesEnabled
 
 - (void)startLocation:(BOOL)enableHighAccuracy
 {
-    if (![self isLocationServicesEnabled]) {
-        [self returnLocationError:PERMISSIONDENIED withMessage:@"Location services are not enabled."];
-        return;
-    }
-    if (![self isAuthorized]) {
-        NSString* message = nil;
-        BOOL authStatusAvailable = [CLLocationManager respondsToSelector:@selector(authorizationStatus)]; // iOS 4.2+
-        if (authStatusAvailable) {
-            NSUInteger code = [CLLocationManager authorizationStatus];
-            if (code == kCLAuthorizationStatusNotDetermined) {
-                // could return POSITION_UNAVAILABLE but need to coordinate with other platforms
-                message = @"User undecided on application's use of location services.";
-            } else if (code == kCLAuthorizationStatusRestricted) {
-                message = @"Application's use of location services is restricted.";
+    [self.commandDelegate runInBackground:^{
+        if (![self isLocationServicesEnabled]) {
```

Comment:
You have to be careful here -- you can't use self in the block, you have to grab a weak reference to self and use it in the block. Like so: https://github.com/apache/cordova-ios/blob/92a62997adb3c8512328d5a0ae42fe5d156291f1/CordovaLib/Classes/Private/Plugins/CDVHandleOpenURL/CDVHandleOpenURL.m#L58

Just us a general note -- I don't notice any of this in your code but it may occur in nested calls -- any calls to UIKit framework classes must occur in the main thread while called inside the block, so that may complicate things.